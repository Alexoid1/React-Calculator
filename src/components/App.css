.title {
  text-align: center;
}

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  height: auto;
  width: auto;
}

.calc-body {
  margin: auto;
}

.card-body {
  margin-top: 30px;
}

.card {
  /* background-color: #e9e7e9; */
  width: 18rem;
  background: #f7f7f7;
  border-radius: 10px 10px 25px 25px;
  box-shadow: inset -15px 15px 15px rgba(255,255,255,.2), 
              inset 15px -15px 15px #e9e7e9, 
              4px 4px 15px rgba(0,0,0,.2), 
              -4px 4px 15px rgba(0,0,0,.2);
}