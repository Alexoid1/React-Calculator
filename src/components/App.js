import React from 'react';
import './App.css';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

class App extends React.Component{
  constructor(props) {
    super(props);

    this.state = {
      total: null,
      next: null,
      operation: null,
    };
    this.handleClick = this.handleClick.bind(this);
    
  }
  handleClick = buttonName => {
    const result = calculate(this.state, buttonName);
    this.setState(result);
  }
  render(){
    return(
      <>
        <h1 className="title">Calculator</h1>
        <Display />
        <ButtonPanel dataState={this.handleClick}/>
      </>
    )
  }
}

export default App;
