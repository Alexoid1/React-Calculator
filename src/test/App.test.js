import React from 'react';
import renderer from 'react-test-renderer';
import App from '../components/App.js'

it('renders correctly the Calculator Page', () => {
    const tree = renderer.create(<App />).toJSON();
  
    expect(tree).toMatchSnapshot();
});