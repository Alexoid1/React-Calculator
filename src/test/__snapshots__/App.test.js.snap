// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly the Calculator Page 1`] = `
<div
  className="card calc-body"
>
  <div
    className="card-body"
  >
    <div
      className="screen mx-auto"
    >
      <div
        className="screen__inner"
      >
        <h3
          className="output"
        >
          
                  
                  
        </h3>
      </div>
    </div>
    <div
      className="card-body buttons-grid"
    >
      <div
        className="row btn--green"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
